rpcs3/Emu/RSX/Program/GLSLSnippets/VideoOutCalibrationPass.glsl

```diff
@@ -0,0 +1,145 @@
+uniform int height;
+#endif
+
+vec4 read_source()
+{
+	if (stereo_display_mode == STEREO_MODE_DISABLED) return texture(fs0, tc0);
+
+	vec4 left, right;
+	if (stereo_image_count == 1)
+	{
+		switch (stereo_display_mode)
+		{
+			case STEREO_MODE_ANAGLYPH_RED_GREEN:
+				left = texture(fs0, tc0 * vec2(1.f, 0.4898f));
+				right = texture(fs0, (tc0 * vec2(1.f, 0.4898f)) + vec2(0.f, 0.510204f));
+				return vec4(left.r, right.g, 0.f, 1.f);
+			case STEREO_MODE_ANAGLYPH_RED_BLUE:
+				left = texture(fs0, tc0 * vec2(1.f, 0.4898f));
+				right = texture(fs0, (tc0 * vec2(1.f, 0.4898f)) + vec2(0.f, 0.510204f));
+				return vec4(left.r, 0.f, right.b, 1.f);
+			case STEREO_MODE_ANAGLYPH_RED_CYAN:
+				left = texture(fs0, tc0 * vec2(1.f, 0.4898f));
+				right = texture(fs0, (tc0 * vec2(1.f, 0.4898f)) + vec2(0.f, 0.510204f));
+				return vec4(left.r, right.g, right.b, 1.f);
+			case STEREO_MODE_ANAGLYPH_MAGENTA_CYAN:
+				left = texture(fs0, tc0 * vec2(1.f, 0.4898f));
+				right = texture(fs0, (tc0 * vec2(1.f, 0.4898f)) + vec2(0.f, 0.510204f));
+				return vec4(left.r, right.g, (left.b + right.b) / 2.f, 1.f);
+			case STEREO_MODE_ANAGLYPH_TRIOSCOPIC:
+				left = texture(fs0, tc0 * vec2(1.f, 0.4898f));
+				right = texture(fs0, (tc0 * vec2(1.f, 0.4898f)) + vec2(0.f, 0.510204f));
+				return vec4(right.r, left.g, right.b, 1.f);
+			case STEREO_MODE_SIDE_BY_SIDE:
+				if (tc0.x < 0.5) return texture(fs0, tc0* sbs_single_matrix);
+				else             return texture(fs0, (tc0* sbs_single_matrix) + vec2(-1.f, 0.510204f));
+			case STEREO_MODE_OVER_UNDER:
+				if (tc0.y < 0.5) return texture(fs0, tc0* ou_single_matrix);
+				else             return texture(fs0, (tc0* ou_single_matrix) + vec2(0.f, 0.020408f) );
+			case STEREO_MODE_INTERLACED:
+				if (mod(height * tc0.y, 2.f) < 1.f) return texture(fs0, tc0 * vec2(1.f, 0.4898f));
+				else                                return texture(fs0, (tc0 * vec2(1.f, 0.4898f)) + vec2(0.f, 0.510204f));
+			default: // undefined behavior
+				return texture(fs0,tc0);
+		}
+	}
+	else if (stereo_image_count == 2)
+	{
+		switch (stereo_display_mode)
+		{
+			case STEREO_MODE_ANAGLYPH_RED_GREEN:
+				left = texture(fs0, tc0);
+				right = texture(fs1, tc0);
+				return vec4(left.r, right.g, 0.f, 1.f);
+			case STEREO_MODE_ANAGLYPH_RED_BLUE:
+				left = texture(fs0, tc0);
+				right = texture(fs1, tc0);
+				return vec4(left.r, 0.f, right.b, 1.f);
+			case STEREO_MODE_ANAGLYPH_RED_CYAN:
+				left = texture(fs0, tc0);
+				right = texture(fs1, tc0);
+				return vec4(left.r, right.g, right.b, 1.f);
+			case STEREO_MODE_ANAGLYPH_MAGENTA_CYAN:
+				left = texture(fs0, tc0);
+				right = texture(fs1, tc0);
+				return vec4(left.r, right.g, (left.b + right.b) / 2.f, 1.f);
+			case STEREO_MODE_ANAGLYPH_TRIOSCOPIC:
+				left = texture(fs0, tc0);
+				right = texture(fs1, tc0);
+				return vec4(right.r, left.g, right.b, 1.f);
+			case STEREO_MODE_SIDE_BY_SIDE:
+				if (tc0.x < 0.5) return texture(fs0,(tc0 * sbs_multi_matrix));
```

Comment:
```suggestion
				if (tc0.x < 0.5) return texture(fs0, (tc0 * sbs_multi_matrix));
```
Same applies for several lines after this one.
The same suggestion to go for a ternary applies but not a deal-breaker.

Reply:
still not my code xD